### ISSUE TITLE
yjs 13.6.4 breaks paths in observed events

### ISSUE THREAD

**Issue by pietrodn** [yjs/yjs#544]
This test case fails with 13.6.4, but succeeds with 13.6.3 and 13.6.1:
```typescript
import Y from 'yjs';
import assert from 'node:assert/strict';

describe('Task bug on YJS', () => {
  it('YMapEvent + YTextEvent paths', () => {
    const doc = new Y.Doc();
    const map1 = doc.getMap('tasks');
    const task1Map = new Y.Map();
    const textNode = new Y.Text('my task label');
    task1Map.set('label', textNode);
    map1.set('task1', task1Map);

    // Listen to events
    const loggedPaths = [];
    map1.observeDeep((events, transaction) => {
      console.log('Begin transaction');
      for (const event of events) {
        const eventType =
          event instanceof Y.YMapEvent ? 'YMapEvent' : event instanceof Y.YTextEvent ? 'YTextEvent' : 'Unknown';
        console.log(`${eventType}, path: ${event.path}`);

        loggedPaths.push(event.path as any);
      }
      console.log('End transaction\n');
    });

    doc.transact(() => {
      textNode.insert(0, 'my task label - ');
      task1Map.set('meta1', new Y.Text('meta1111'));
    });

    // On 13.6.4, loggedPaths == [[], ['label']]
    assert.deepEqual(loggedPaths, [['task1'], ['task1', 'label']]);
  });
});
```

My hypothesis is the following: in 13.6.4, `YEvent.path` getter was changed to cache the value. However, in `Transaction.js` a line of codes [mutates](https://github.com/yjs/yjs/blob/ce098d0ac2b617d0f4c5214c95607dee8cfaba87/src/utils/Transaction.js#L289) `YEvent.currentTarget`, one of the parameters used to compute the path. So, it is not possible to cache that value safely.

The bug happens when sibling `YMap` and `YText` are changed in the same transaction.

**To Reproduce**
Run the test first on 13.6.3 and verify that it passes.
Run it again on 13.6.4, and it should fail on the final assert.

**Expected behavior**
The test should pass on 13.6.4.

**Environment Information**
 - NodeJS 18.16.0
 - YJS 13.6.4

**Comment by ulion** [comment]
Wow! Your exceptional work in identifying that elusive bug has truly helped the 13.6.4 victims. We greatly appreciate all of your efforts.

**Comment by dmonad** [comment]
Thank you for investigating this issue @pietrodn !! 

I merged @NilSet fix #545 and will make a new release shortly.
